Documentation/Input/Pointers.md

```diff
@@ -1,39 +1,239 @@
+        }
+
+        private void IMixedRealityPointerHandler.OnPointerDown(MixedRealityPointerEventData eventData) {}
+
+        private void IMixedRealityPointerHandler.OnPointerDragged(MixedRealityPointerEventData eventData) { }
+
+        private void IMixedRealityPointerHandler.OnPointerClicked(MixedRealityPointerEventData eventData) 
+        {
+            material.color = color_OnSelect;
+        }
+```
+
+#### Register globally
+
+blah need to add code example for CoreServices.InputSystem.Register etc
+
 Alternatively, there's the possibility to derive the handler script from [`InputSystemGlobalHandlerListener`](xref:Microsoft.MixedReality.Toolkit.Input.InputSystemGlobalHandlerListener) for global registration or from [`BaseInputHandler`](xref:Microsoft.MixedReality.Toolkit.Input.BaseInputHandler) for choosing in the inspector whether to register as global listener or not. In both cases the abstract methods `RegisterHandlers` and `UnregisterHandlers` which specify the interfaces to listen to, need to be implemented.
 
-For pointer events, i.e. the ones handled by [`IMixedRealityPointerHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealityPointerHandler), we provide further convenience in the form of [`PointerHandler`](xref:Microsoft.MixedReality.Toolkit.Input.PointerHandler). This script allows you to handle pointer events directly via Unity Events.
+### Query Pointers
+
+It is possible to gather all pointers currently active by looping through the available input sources (i.e controllers and inputs available) to discover which pointers are attached to them.
+
+```csharp
+    HashSet<IMixedRealityPointer> pointers = new HashSet<IMixedRealityPointer>();
+
+    // Find all valid pointers
+    foreach (var inputSource in CoreServices.InputSystem.DetectedInputSources)
+    {
+        foreach (var pointer in inputSource.Pointers)
+        {
+            if (pointer.IsInteractionEnabled && !pointers.Contains(pointer))
+            {
+                pointers.Add(pointer);
+            }
+        }
+    }
+```
+
+#### Primary Pointer
+
+Developers can subscribe to the FocusProviders PrimaryPointerChanged event to be notified when the primary pointer in focus has changed. This can be extremely useful to identify if the user is currently interacting with a scene via gaze or a hand ray or other input source.
 
-<img src="../../Documentation/Images/Input/PointerHandler.png" style="max-width:100%;">
+```csharp
+    private void OnEnable()
+    {
+        CoreServices.InputSystem?.FocusProvider?.SubscribeToPrimaryPointerChanged(OnPrimaryPointerChanged, true);
+    }
 
-## Pointer Result
+    private void OnPrimaryPointerChanged(IMixedRealityPointer oldPointer, IMixedRealityPointer newPointer)
+    {
+        ...
+    }
+
+    private void OnDisable()
+    {
+        CoreServices.InputSystem?.FocusProvider?.UnsubscribeFromPrimaryPointerChanged(OnPrimaryPointerChanged);
+        OnPrimaryPointerChanged(null, null);
```

Comment:
>OnPrimaryPointerChanged(null, null); [](start = 8, length = 36)

Nit: Add a comment stating that this is designed to flush out the current primary pointer